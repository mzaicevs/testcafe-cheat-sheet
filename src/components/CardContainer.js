import React from "react";
import { CardColumns, InputGroup, Form, Navbar } from "react-bootstrap";
import search_mappings from "./search/search_mappings";

const Search = ({ handleChange }) => (
  <InputGroup style={{ marginTop: "12px", marginBottom: "12px" }}>
    <InputGroup.Prepend>
      <InputGroup.Text id="inputGroupPrepend">
        <span role="img" aria-label="search">
          ðŸ”Ž
        </span>
      </InputGroup.Text>
    </InputGroup.Prepend>
    <Form.Control
      type="text"
      placeholder="Search for ..."
      aria-label="inputGroupPrepend"
      onChange={e => handleChange(e.target.value)}
    />
  </InputGroup>
);

class CardContainer extends React.Component {
  constructor() {
    super();
    this.state = {
      filterValue: ""
    };
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(value) {
    this.setState({ filterValue: value });
  }

  render() {
    const filteredSearchComponents = search_mappings.filter(mapping => {
      let foundValue = false;
      for (const keyword of mapping.keywords) {
        if (keyword.includes(this.state.filterValue)) {
          foundValue = true;
          break;
        }
      }
      return foundValue;
    });

    const renderSearchComponents = mappingArray =>
      mappingArray.map(mapping => mapping.getComponent());

    return (
      <>
        <CardColumns>
          {this.state.filterValue
            ? // mapped only when there is a filter value
              renderSearchComponents(filteredSearchComponents)
            : // map everything by default
              renderSearchComponents(search_mappings)}
        </CardColumns>
        <Navbar bg="light" variant="light" fixed="bottom">
          <Navbar.Brand href="#home">TestCafe Cheat Sheet</Navbar.Brand>
          <Search handleChange={this.handleChange} />
        </Navbar>
      </>
    );
  }
}

export default CardContainer;
