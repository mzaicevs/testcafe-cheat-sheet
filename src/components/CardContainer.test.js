import React from "react";
import { shallow } from "enzyme";
import CardContainer from "./CardContainer";

describe("CardContainer", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<CardContainer />);
    expect(wrapper).toMatchSnapshot();
  });
});
