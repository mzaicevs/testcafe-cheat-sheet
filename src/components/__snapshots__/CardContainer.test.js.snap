// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer matches snapshot 1`] = `
<Fragment>
  <CardColumns>
    <Loading
      component={<UNDEFINED />}
      key="1"
      spinnerAnimation="grow"
    />
    <Loading
      component={<UNDEFINED />}
      key="2"
      spinnerAnimation="grow"
    />
    <Loading
      component={<UNDEFINED />}
      key="3"
      spinnerAnimation="grow"
    />
    <Loading
      component={<UNDEFINED />}
      key="4"
      spinnerAnimation="grow"
    />
    <Loading
      component={<UNDEFINED />}
      key="5"
      spinnerAnimation="grow"
    />
    <Loading
      component={<UNDEFINED />}
      key="6"
      spinnerAnimation="grow"
    />
    <Loading
      component={<UNDEFINED />}
      key="7"
      spinnerAnimation="grow"
    />
    <Loading
      component={<UNDEFINED />}
      key="8"
      spinnerAnimation="grow"
    />
    <Loading
      component={<UNDEFINED />}
      key="9"
      spinnerAnimation="grow"
    />
    <Loading
      component={<UNDEFINED />}
      key="10"
      spinnerAnimation="grow"
    />
    <Loading
      component={<UNDEFINED />}
      key="11"
      spinnerAnimation="grow"
    />
    <Loading
      component={<UNDEFINED />}
      key="12"
      spinnerAnimation="grow"
    />
    <Loading
      component={<UNDEFINED />}
      key="13"
      spinnerAnimation="grow"
    />
    <Loading
      component={<UNDEFINED />}
      key="14"
      spinnerAnimation="grow"
    />
  </CardColumns>
  <Bootstrap(undefined)
    bg="light"
    fixed="bottom"
    variant="light"
  >
    <Search
      handleChange={[Function]}
    />
  </Bootstrap(undefined)>
</Fragment>
`;
